l10n_it_delivery_note/models/account_invoice.py

```diff
@@ -157,3 +157,13 @@ def unlink(self):
         for dn in dnls_to_unlink.mapped("delivery_note_id"):
             dn.state = "confirm"
         return res
+
+    def button_cancel(self):  # pylint: disable=missing-return
+        super().button_cancel()
+        dn_lines = (
+            self.invoice_line_ids.sale_line_ids.delivery_note_line_ids
+            | self.delivery_note_ids.line_ids
+        )
+        dn_lines.sync_invoice_status()
+        dn_lines.delivery_note_id._compute_invoice_status()
+        dn_lines.delivery_note_id.state = "confirm"
```

Comment:
**chore(non-blocking)** credo sia comunque una buona pratica per questioni di compatibilità fare sempre

```py
res = super().button_cancel()
...
return res
```

e così si può togliere il commento di pylint

Reply:
Non mi piace, dà l'impressione che si sia un valore di ritorno, che in realtà non c'è. 

https://github.com/odoo/odoo/blob/649a7f185dca806af3b3d54a1e4390b9baff79c6/addons/account/models/account_move.py#L3896

Se il warning di pylint è sbagliato (e in questo caso lo è, la super() la fai in metodo privo di return), lo disabiliti, piuttosto che modificare il tuo codice (giusto) per far contento pylint.

https://github.com/OCA/queue/blob/20d343613ea26c6a4eacbff4b5b0aa81c7817b81/queue_job/fields.py#L43